src/test/java/uk/gov/hmcts/reform/civil/handler/event/EvidenceUploadNotificationEventHandlerTest.java

```diff
@@ -5,26 +5,82 @@
 import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
 import uk.gov.hmcts.reform.civil.event.EvidenceUploadNotificationEvent;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.notification.EvidenceUploadApplicantNotificationHandler;
+import uk.gov.hmcts.reform.civil.notification.EvidenceUploadRespondentNotificationHandler;
+import uk.gov.hmcts.reform.civil.sampledata.CaseDataBuilder;
+import uk.gov.hmcts.reform.civil.sampledata.CaseDetailsBuilder;
 import uk.gov.hmcts.reform.civil.service.CoreCaseDataService;
 
+import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.verify;
-import static uk.gov.hmcts.reform.civil.callback.CaseEvent.EVIDENCE_UPLOAD_NOTIFICATION;
+import static org.mockito.Mockito.when;
 
 @ExtendWith(SpringExtension.class)
 class EvidenceUploadNotificationEventHandlerTest {
 
     @Mock
     private CoreCaseDataService coreCaseDataService;
 
+    @Mock
+    private CaseDetailsConverter caseDetailsConverter;
+    @Mock
+    private EvidenceUploadApplicantNotificationHandler applicantNotificationHandler;
+    @Mock
+    private EvidenceUploadRespondentNotificationHandler respondentNotificationHandler;
     @InjectMocks
     private EvidenceUploadNotificationEventHandler handler;
 
     @Test
-    void shouldCallTriggerEventWithExpectedParams_WhenEvidenceUploadEvent() {
-        EvidenceUploadNotificationEvent event = new EvidenceUploadNotificationEvent(1L);
+    void shouldNotifyAllSolicitors() {
+        Long caseId = 1633357679902210L;
+        CaseData caseData = new CaseDataBuilder().atStateClaimDraft()
+            .build();
+        CaseDetails caseDetails = CaseDetailsBuilder.builder().data(caseData).build();
+        when(coreCaseDataService.getCase(caseId)).thenReturn(caseDetails);
+        when(caseDetailsConverter.toCaseData(caseDetails)).thenReturn(caseData);
+        EvidenceUploadNotificationEvent event = new EvidenceUploadNotificationEvent(caseId);
+        handler.sendEvidenceUploadNotification(event);
+
+        verify(applicantNotificationHandler).notifyApplicantEvidenceUpload(caseData);
+        verify(respondentNotificationHandler).notifyRespondentEvidenceUpload(caseData, false);
+        verify(respondentNotificationHandler).notifyRespondentEvidenceUpload(caseData, true);
```

Comment:
Add `//Given`, `//When`, `//Then` sections to the tests for clarity.

Reply:
done